util/conditions/getter.go

```diff
@@ -0,0 +1,99 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package conditions
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	clusterv1 "sigs.k8s.io/cluster-api/api/v1alpha3"
+)
+
+// Getter interface defines methods that a Cluster API object should implement in order to
+// use the conditions package for getting conditions.
+type Getter interface {
```

Comment:
Any specific reason to use `Getter/Setter` instead of the more generic `Reader/Writer`?

Reply:
I find `Getter/Setter` more appropriate having `GetConditions/SetConditions` methods, but I don't have strong opinions here and I'm happy to change if preferred